engine/src/main/scala/cromwell/server/CromwellSystem.scala

```diff
@@ -34,11 +34,15 @@ trait CromwellSystem {
   implicit private final lazy val ec = actorSystem.dispatcher
 
   def shutdownActorSystem(): Future[Terminated] = {
-    Http().shutdownAllConnectionPools() flatMap { _ =>
-      shutdownMaterializerAndActorSystem()
-    } recoverWith {
-      case _ => shutdownMaterializerAndActorSystem()
-    }
+    // If the actor system is already terminated it's already too late for a clean shutdown
+    if (!actorSystem.whenTerminated.isCompleted) {
+      Http().shutdownAllConnectionPools() flatMap { _ =>
+        shutdownMaterializerAndActorSystem()
+      } recoverWith {
+        case _ =>
+          shutdownMaterializerAndActorSystem()
```

Comment:
I see it's preexisting but is this `recover` in the right place? Try to run `shutdownMaterializerAndActorSystem` and if that doesn't work run it again? If so and you understand why it's like that a comment would be appreciated. 🙂